subprojects/packagefiles/libtommath/meson.build

```
@@ -0,0 +1,380 @@
+  'bn_mp_grow.c',
+  'bn_mp_incr.c',
+  'bn_mp_init.c',
+  'bn_mp_init_copy.c',
+  'bn_mp_init_i32.c',
+  'bn_mp_init_i64.c',
+  'bn_mp_init_l.c',
+  'bn_mp_init_multi.c',
+  'bn_mp_init_set.c',
+  'bn_mp_init_size.c',
+  'bn_mp_init_u32.c',
+  'bn_mp_init_u64.c',
+  'bn_mp_init_ul.c',
+  'bn_mp_invmod.c',
+  'bn_mp_is_square.c',
+  'bn_mp_iseven.c',
+  'bn_mp_isodd.c',
+  'bn_mp_kronecker.c',
+  'bn_mp_lcm.c',
+  'bn_mp_log_n.c',
+  'bn_mp_lshd.c',
+  'bn_mp_mod.c',
+  'bn_mp_mod_2d.c',
+  'bn_mp_mod_d.c',
+  'bn_mp_montgomery_calc_normalization.c',
+  'bn_mp_montgomery_reduce.c',
+  'bn_mp_montgomery_setup.c',
+  'bn_mp_mul.c',
+  'bn_mp_mul_2.c',
+  'bn_mp_mul_2d.c',
+  'bn_mp_mul_d.c',
+  'bn_mp_mulmod.c',
+  'bn_mp_neg.c',
+  'bn_mp_or.c',
+  'bn_mp_pack.c',
+  'bn_mp_pack_count.c',
+  'bn_mp_prime_fermat.c',
+  'bn_mp_prime_frobenius_underwood.c',
+  'bn_mp_prime_is_prime.c',
+  'bn_mp_prime_miller_rabin.c',
+  'bn_mp_prime_next_prime.c',
+  'bn_mp_prime_rabin_miller_trials.c',
+  'bn_mp_prime_rand.c',
+  'bn_mp_prime_strong_lucas_selfridge.c',
+  'bn_mp_radix_size.c',
+  'bn_mp_radix_smap.c',
+  'bn_mp_rand.c',
+  'bn_mp_read_radix.c',
+  'bn_mp_reduce.c',
+  'bn_mp_reduce_2k.c',
+  'bn_mp_reduce_2k_l.c',
+  'bn_mp_reduce_2k_setup.c',
+  'bn_mp_reduce_2k_setup_l.c',
+  'bn_mp_reduce_is_2k.c',
+  'bn_mp_reduce_is_2k_l.c',
+  'bn_mp_reduce_setup.c',
+  'bn_mp_root_n.c',
+  'bn_mp_rshd.c',
+  'bn_mp_sbin_size.c',
+  'bn_mp_set.c',
+  'bn_mp_set_double.c',
+  'bn_mp_set_i32.c',
+  'bn_mp_set_i64.c',
+  'bn_mp_set_l.c',
+  'bn_mp_set_u32.c',
+  'bn_mp_set_u64.c',
+  'bn_mp_set_ul.c',
+  'bn_mp_shrink.c',
+  'bn_mp_signed_rsh.c',
+  'bn_mp_sqr.c',
+  'bn_mp_sqrmod.c',
+  'bn_mp_sqrt.c',
+  'bn_mp_sqrtmod_prime.c',
+  'bn_mp_sub.c',
+  'bn_mp_sub_d.c',
+  'bn_mp_submod.c',
+  'bn_mp_to_radix.c',
+  'bn_mp_to_sbin.c',
+  'bn_mp_to_ubin.c',
+  'bn_mp_ubin_size.c',
+  'bn_mp_unpack.c',
+  'bn_mp_xor.c',
+  'bn_mp_zero.c',
+  'bn_prime_tab.c',
+  'bn_s_mp_add.c',
+  'bn_s_mp_balance_mul.c',
+  'bn_s_mp_div_3.c',
+  'bn_s_mp_exptmod.c',
+  'bn_s_mp_exptmod_fast.c',
+  'bn_s_mp_get_bit.c',
+  'bn_s_mp_invmod_fast.c',
+  'bn_s_mp_invmod_slow.c',
+  'bn_s_mp_karatsuba_mul.c',
+  'bn_s_mp_karatsuba_sqr.c',
+  'bn_s_mp_log.c',
+  'bn_s_mp_log_2expt.c',
+  'bn_s_mp_log_d.c',
+  'bn_s_mp_montgomery_reduce_fast.c',
+  'bn_s_mp_mul_digs.c',
+  'bn_s_mp_mul_digs_fast.c',
+  'bn_s_mp_mul_high_digs.c',
+  'bn_s_mp_mul_high_digs_fast.c',
+  'bn_s_mp_prime_is_divisible.c',
+  'bn_s_mp_rand_jenkins.c',
+  'bn_s_mp_rand_platform.c',
+  'bn_s_mp_reverse.c',
+  'bn_s_mp_sqr.c',
+  'bn_s_mp_sqr_fast.c',
+  'bn_s_mp_sub.c',
+  'bn_s_mp_toom_mul.c',
+  'bn_s_mp_toom_sqr.c',
+)
+
+inc = include_directories('.')
+
+# Add extra optimization flags to match Makefile and CMake builds
+# Note: Meson already adds -O3 when optimization=3 (from buildtype=release)
+# We only need to add the flags Meson doesn't add by default
+if get_option('buildtype') != 'debug'
```

This will not work properly on all Meson versions.
```suggestion
if not get_option('debug')
```